app/templating/view_context.py

```diff
@@ -92,24 +95,32 @@ def build_view_context_for_list_collector(
     question_context = build_view_context_for_question(rendered_block, form)
 
     list_name = rendered_block['populates_list']
-    list_item_ids = list_store[list_name]
+    list_item_ids = list_store[list_name].list_items
 
     list_title_answer_ids = [
         answer['id'] for answer in rendered_block['add_block']['question']['answers']
     ]
 
+    primary_person = list_store.get_primary_person(list_name)
```

Comment:
We need to review the use of context for the primary person. The list store abstracts it into a context, but here we're adding primary person specific logic to the view context.